functions/grapher/[slug].ts

```diff
@@ -27,10 +24,16 @@ router
     )
     .all("*", () => error(404, "Route not defined"))
 
+export const onRequestOptions: PagesFunction = async (_context) => {
+    return new Response(null, {
+        status: 204,
+        headers: {
+            "Access-Control-Allow-Origin": "*",
+        },
+    })
+}
+
 export const onRequestGet: PagesFunction = async (context) => {
-    // Makes it so that if there's an error, we will just deliver the original page before the HTML rewrite.
-    // Only caveat is that redirects will not be taken into account for some reason; but on the other hand the worker is so simple that it's unlikely to fail.
-    context.passThroughOnException()
```

Comment:
We should really try to keep this in as the first line of business.

It allows us that if any unhandled exception happens in this handler, we're still _failing open_ and are sending over the unmodified `/grapher/...` response as-is from the origin server. Which is a good fallback (and will work for graphers unless there's a redirect - but never for `.json`, `.png` and `.svg`, obviously.)

See https://developers.cloudflare.com/workers/runtime-apis/context/#passthroughonexception.